packages/webview_flutter/webview_flutter_wkwebview/lib/src/webview_cupertino_widget.dart

```diff
@@ -0,0 +1,227 @@
+    required CreationParams creationParams,
+    required this.callbacksHandler,
+    required this.javascriptChannelRegistry,
+    web_kit.WebViewConfiguration? configuration,
+    FutureOr<String>? systemVersion,
+    @visibleForTesting this.webViewProxy = const WebViewProxy(),
+  }) : super(callbacksHandler) {
+    this.systemVersion = systemVersion ?? _defaultSystemVersion;
+    _setCreationParams(
+      creationParams,
+      configuration: configuration ?? web_kit.WebViewConfiguration(),
+    ).then((_) => _initializationCompleter.complete());
+  }
+
+  // Since the version of iOS won't change once an app is started, this only
+  // needs to retrieved for the first WebView.
+  static final Future<IosDeviceInfo> _defaultDeviceInfo =
+      DeviceInfoPlugin().iosInfo;
+
+  static Future<String> get _defaultSystemVersion async {
+    final String? systemVersion = (await _defaultDeviceInfo).systemVersion;
+    assert(
+      systemVersion != null,
+      'Unable to retrieve the system version of this device.',
+    );
+    return systemVersion!;
+  }
+
+  final Completer<void> _initializationCompleter = Completer<void>();
+
+  /// Handles callbacks that are made by navigation.
+  final WebViewPlatformCallbacksHandler callbacksHandler;
+
+  /// Manages named JavaScript channels and forwarding incoming messages on the correct channel.
+  final JavascriptChannelRegistry javascriptChannelRegistry;
+
+  /// Handles constructing a [web_kit.WebView].
+  ///
+  /// This should only be changed when used for testing.
+  final WebViewProxy webViewProxy;
+
+  /// Represents the WebView maintained by platform code.
+  late final web_kit.WebView webView;
+
+  /// The system version of the iOS device.
+  late final FutureOr<String> systemVersion;
+
+  Future<void> _initialize() => _initializationCompleter.future;
+
+  Future<void> _setCreationParams(
+    CreationParams params, {
+    required web_kit.WebViewConfiguration configuration,
+  }) async {
+    await _setWebViewConfiguration(
+      configuration,
+      allowsInlineMediaPlayback: params.webSettings?.allowsInlineMediaPlayback,
+      autoMediaPlaybackPolicy: params.autoMediaPlaybackPolicy,
+    );
+
+    webView = webViewProxy.createWebView(configuration);
+  }
+
+  Future<void> _setWebViewConfiguration(
+    web_kit.WebViewConfiguration configuration, {
+    required bool? allowsInlineMediaPlayback,
+    required AutoMediaPlaybackPolicy autoMediaPlaybackPolicy,
+  }) async {
+    if (allowsInlineMediaPlayback != null) {
+      configuration.allowsInlineMediaPlayback = allowsInlineMediaPlayback;
+    }
+
+    late final bool requiresUserAction;
+    switch (autoMediaPlaybackPolicy) {
+      case AutoMediaPlaybackPolicy.require_user_action_for_all_media_types:
+        requiresUserAction = true;
+        break;
+      case AutoMediaPlaybackPolicy.always_allow:
+        requiresUserAction = false;
+        break;
+    }
+
+    if (_isGreaterOrEqualVersion(await systemVersion, '10.0')) {
```

Comment:
An alternative to having this version check logic would be to only support `mediaTypesRequiringUserActionForPlayback` and handle calling the other methods with platform code. However, I wanted to try a solution of handling version conflicts in Dart. A future feature may require a more complex solution than a 1 liner like this one.